### PR TITLE
cloudbuild: Use commit SHA instead of latest to trigger redeployment

### PR DESCRIPTION
**Pull request by lotas**
Problem with using ":latest" is that deployment configuration doesn't change, so the pods are not being restarted

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -12,20 +12,23 @@ steps:
       - DOCKER_FLOW_VERSION=${_VERSION}
       - '-t'
       - ${_IMAGE_NAME}:${_IMAGE_VERSION}
+      - '-t'
+      - ${_IMAGE_NAME}:latest
```
Comment (lotas) [inline cloudbuild.yaml:16]: adding `:latest` tag to make sure any dev can use `gcr.io/taskcluster-dev/taskcluster-dev/main:latest` for their needs

File: cloudbuild.yaml [hunk 2]
```diff
@@ -12,20 +12,23 @@ steps:
...
       - .
     id: Build
   - name: gcr.io/cloud-builders/docker
     args:
       - push
-      - ${_IMAGE_NAME}:${_IMAGE_VERSION}
+      - '-a'
+      - ${_IMAGE_NAME}
```
Comment (lotas) [inline cloudbuild.yaml:23]: pushing both tags

File: cloudbuild.yaml [hunk 3]
```diff
@@ -12,20 +12,23 @@ steps:
...
     id: Push
   - name: gcr.io/cloud-builders/docker
     id: Build deploy image using cache from previous build
     entrypoint: bash
     args:
       - '-c'
       - |
-        docker build -t ${_DEPLOY_IMAGE_NAME}:${_IMAGE_VERSION} --cache-from ${_DEPLOY_IMAGE_NAME}:latest -<<EOF
+        docker build -t ${_DEPLOY_IMAGE_NAME}:latest --cache-from ${_DEPLOY_IMAGE_NAME}:latest -<<EOF
```
Comment (lotas) [inline cloudbuild.yaml:31]: deploy helper image is always the same, so single `:latest` tag for it is enough

File: cloudbuild.yaml [hunk 4]
```diff
@@ -78,8 +82,7 @@ steps:
       - TASKCLUSTER_ACCESS_TOKEN
 timeout: 1800s
 images:
-  - ${_IMAGE_NAME}:${_IMAGE_VERSION}
-  - ${_DEPLOY_IMAGE_NAME}:${_IMAGE_VERSION}
+  - ${_DEPLOY_IMAGE_NAME}:latest
```
Comment (lotas) [inline cloudbuild.yaml:85]: we only allow gcloud build to push this deploy image automatically, as we are pushing main monoimage manually already

**Review by matt-boris — APPROVED** [review]
LGTM, thanks!

**Pull request merged**